src/Mvc/Mvc.ApiExplorer/test/EndpointMetadataApiDescriptionProviderTest.cs

```diff
@@ -111,14 +111,14 @@ public void AddsMultipleRequestFormatsFromMetadata()
         public void AddsMultipleRequestFormatsFromMetadataWithRequestType()
         {
             var apiDescription = GetApiDescription(
-                [Consumes(typeof(InferredJsonClass), "application/custom0", "application/custom1")]
+                [Consumes(typeof(InferredJsonClass), "application/custom0", "application/custom1", IsOptional = true)]
                 () => { });
 
             Assert.Equal(2, apiDescription.SupportedRequestFormats.Count);
 
             var apiParameterDescription = apiDescription.ParameterDescriptions[0];
             Assert.Equal("InferredJsonClass", apiParameterDescription.Type.Name);
-            Assert.True(apiParameterDescription.IsRequired);
+            Assert.False(apiParameterDescription.IsRequired);
```

Comment:
Is there a test that verifies this is `true` if `IsOptional` is not configured?

Reply:
Lets discuss this later. I have a concern with the ```isOptional``` for ```ConsumesAttribute``` when you don't use a type. The following case might have some unintended behaviors because of the inferred type: 

 ```C#
    app.MapPost("/todo", [Consumes("application/custom0", "application/custom1")] (Todo todo) => {...})
```

Reply:
so when type is null, we assume the type information comes from the body parameter, and therefore we can use the optionality from there?

Reply:
@davidfowl That's correct. Any unintended consequence that this change can cause that you may know of ?.